pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -312,40 +316,77 @@ func (plugin *cniNetworkPlugin) TearDownPod(namespace string, name string, id ku
 	return plugin.deleteFromNetwork(plugin.getDefaultNetwork(), name, namespace, id, netnsPath, nil)
 }
 
-func podDesc(namespace, name string, id kubecontainer.ContainerID) string {
+func PodDesc(namespace, name string, id kubecontainer.ContainerID) string {
 	return fmt.Sprintf("%s_%s/%s", namespace, name, id.ID)
 }
 
+func GetIpFromResult(res *cnicurrent.Result, rt *libcni.RuntimeConf) string {
+	klog.V(4).Infof("res: %s", res.String())
+	rt_json, err := json.Marshal(rt)
+	if err == nil {
+		klog.V(4).Infof("rt: %s", string(rt_json))
+	}
+
+	for _, ip := range res.IPs {
+		if ip.Interface == nil || *ip.Interface > len(res.Interfaces) {
+			continue
+		}
+		intf := res.Interfaces[*ip.Interface]
+		if intf.Sandbox != "" && intf.Name == rt.IfName {
+			// Found our sandbox interface, use the IP
+			return ip.Address.IP.String()
+		}
+	}
+	// Fall back to first un-associated address for compatibility
+	// with older CNI plugins
+	klog.Warningf("There is no ip address associated with network interface : %s", rt.IfName)
+	return res.IPs[0].Address.IP.String()
+}
+
 func (plugin *cniNetworkPlugin) addToNetwork(network *cniNetwork, podName string, podNamespace string, podSandboxID kubecontainer.ContainerID, podNetnsPath string, annotations, options map[string]string) (cnitypes.Result, error) {
 	rt, err := plugin.buildCNIRuntimeConf(podName, podNamespace, podSandboxID, podNetnsPath, annotations, options)
 	if err != nil {
 		klog.Errorf("Error adding network when building cni runtime conf: %v", err)
 		return nil, err
 	}
 
-	pdesc := podDesc(podNamespace, podName, podSandboxID)
+	pdesc := PodDesc(podNamespace, podName, podSandboxID)
 	netConf, cniNet := network.NetworkConfig, network.CNIConfig
 	klog.V(4).Infof("Adding %s to network %s/%s netns %q", pdesc, netConf.Plugins[0].Network.Type, netConf.Name, podNetnsPath)
-	res, err := cniNet.AddNetworkList(netConf, rt)
+	res, err := cniNet.AddNetworkList(context.TODO(), netConf, rt)
 	if err != nil {
 		klog.Errorf("Error adding %s to network %s/%s: %v", pdesc, netConf.Plugins[0].Network.Type, netConf.Name, err)
 		return nil, err
 	}
 	klog.V(4).Infof("Added %s to network %s: %v", pdesc, netConf.Name, res)
+
+	if res == nil {
```

Comment:
You'll never get a nil result if err is also nil. I think we can skip this check.